### PR TITLE
Add  Interview Slot

### LINKED ISSUE #38: Add Interview Date Field

### PR DESCRIPTION
**Pull request by laughingkid-sg**
Fix #38

### CODE REVIEW

**Comment by laughingkid-sg** [comment]
# Added Interview Slot field in Application

## Key Summary
- Created  `InterviewSlot` class
- Interview slot is stored as a `LocalDateTime` value in the `InterviewSlot` class
- Adding a new application will default the value to `LocalDateTime.Max` & the value will default to **_Interview date is not set._**
- Edit command is re-enabled to cater to the adding of the interview time slot
-  Comparison between `InterviewSlot` class compares its `String` value instead of its actual value because when saving to JSON, data of  `LocalDateTime.Max` is truncated
- Input format is `dd-MM-yyyy HH:mm` and display format is `d MMM yyyy HH:mm`

## Usage

### Adding/Replacing interview slot
Command: `edit 1 idt/13-03-2022 13:00`
Feedback: 

> Edited Person: Shopee; Job Title: Software Engineer Intern; Phone: 87438807; Email: hr@shopee.sg; Address: 5 Science Park Dr, #06-40; Interview Slot: 13 Mar 2022 13:00; Tags: [friends]

### Removing interview slot
Command: `edit 1 idt/`
Feedback: 

> Edited Person: Shopee; Job Title: Software Engineer Intern; Phone: 87438807; Email: hr@shopee.sg; Address: 5 Science Park Dr, #06-40; Interview Slot: Interview date is not set.; Tags: [friends]

## Preview
image

## Other Updates
- Update `SampleDataUtil` to reflect applications 
- Updated some test case data to reflect applications


## Further improvements
1. Will highlight in GUI that interview date is not set for application depending on tags
2. Additional test cases for InterviewSlot
3. Suggestion to change `idt/` to 'interview/'
4. Update remaining test cases data to reflect the application

Ready for review @lchokhoe @tanyutao544 @joszx

File: src/test/java/seedu/address/logic/parser/CommandParserTestUtil.java [hunk 1]
```diff
@@ -16,6 +16,7 @@
      */
     public static void assertParseSuccess(Parser parser, String userInput, Command expectedCommand) {
         try {
+
```
Comment (lchokhoe) [inline src/test/java/seedu/address/logic/parser/CommandParserTestUtil.java]: You can remove this line as well while you are reverting the changes to lchokhoe.png.

File: src/test/java/seedu/address/testutil/PersonBuilder.java [hunk 2]
```diff
@@ -20,16 +23,21 @@
     public static final String DEFAULT_NAME = "Amy Bee";
     public static final String DEFAULT_PHONE = "85355255";
     public static final String DEFAULT_EMAIL = "amy@gmail.com";
+    public static final String DEFAULT_INTERVIEWSLOT = LocalDateTime.MAX.format(DateTimeFormatter
+            .ofPattern(InterviewSlot.FORMAT_DATETIME_INPUT));
     public static final String DEFAULT_ADDRESS = "123, Jurong West Ave 6, #08-111";
     public static final String DEFAULT_JOBTITLE = "Intern";
 
+
```
Comment (lchokhoe) [inline src/test/java/seedu/address/testutil/PersonBuilder.java]: You can remove this line as well while you are reverting the changes to lchokhoe.png.

File: src/test/java/seedu/address/testutil/PersonBuilder.java [hunk 3]
```diff
@@ -20,16 +23,21 @@
...
     private Name name;
     private Phone phone;
     private Email email;
+    private InterviewSlot interviewSlot;
     private Address address;
     private Set<Tag> tags;
     private JobTitle jobTitle;
 
+
```
Comment (lchokhoe) [inline src/test/java/seedu/address/testutil/PersonBuilder.java]: You can remove this line as well while you are reverting the changes to lchokhoe.png.

File: src/main/java/seedu/address/model/application/InterviewSlot.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
+package seedu.address.model.application;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+
```
Comment (lchokhoe) [inline src/main/java/seedu/address/model/application/InterviewSlot.java]: I think you can clear away these extra lines while you are reverting the changes to lchokhoe.png.

**Review by lchokhoe — CHANGES_REQUESTED** [review]
**lchokhoe.png file does not need to be updated, please revert the changes in another commit.**

While you are at it, some of the files had extra lines as well, I commented on them but here's where they are:
1. (main folder) InterviewSlot.java (Line 10 - Line 13)
2. (test folder) CommandParserTestUtil.java (Line 19)
3. (test folder) PersonBuilder.java (Line 31 and Line 40)

Other then that I think all the changes you made LGTM 👍

**Comment by laughingkid-sg** [comment]
> **lchokhoe.png file does not need to be updated, please revert the changes in another commit.**
> 
> While you are at it, some of the files had extra lines as well, I commented on them but here's where they are:
> 
> 1. (main folder) InterviewSlot.java (Line 10 - Line 13)
> 2. (test folder) CommandParserTestUtil.java (Line 19)
> 3. (test folder) PersonBuilder.java (Line 31 and Line 40)
> 
> Other then that I think all the changes you made LGTM 👍

@lchokhoe Image updated as per #59 & remove blank spaces

**Review by lchokhoe — APPROVED** [review]
Nice!

**Pull request merged**